### ISSUE TITLE
ES6 - no-extra-parens on callback function

### ISSUE THREAD

**Issue by jessereitsma** [eslint/eslint#8482]
**Tell us about your environment**

* **ESLint Version:** v3.19.0
* **Node Version:** v7.8.0
* **npm Version:** v4.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**
standard.js

**Please show your full configuration:**

```
module.exports = {
  root: true,
  extends: 'standard',
  plugins: [
    'html', 'standard', 'promise'
  ],
  'rules': {
    'arrow-parens': 0,
    'semi': 0,
    'no-extra-semi': 0,
    'no-debugger': process.env.NODE_ENV === 'production' ? 2 : 0
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
query (sql, params) {
  connection.query(sql, params, ((error, results) => {
    // Callback for connection.query
  });
}
```

**What did you expect to happen?**
To not get the following error:
```
error  Gratuitous parentheses around expression  no-extra-parens
```
**What actually happened? Please include the actual, raw output from ESLint.**
```
error  Gratuitous parentheses around expression  no-extra-parens
```

The callback I provided seems valid to me.

**Comment by jessereitsma** [comment]
As a workaround I've changed my code to use bluebird's `promisify`.
That will get rid of the callback, and also the error.
But I still think this isn't the expected behaviour.

**Comment by platinumazure** [comment]
I think your example is missing a parenthesis. I assume this is what you meant?

```js
query (sql, params) {
  connection.query(sql, params, ((error, results) => {
    // Callback for connection.query
  }));  // <-- added right paren here
}
```

If so, you can definitely remove the parentheses around the arrow function:

```js
query (sql, params) {
  connection.query(sql, params, (error, results) => {
    // Callback for connection.query
  });
}
```

**Comment by jessereitsma** [comment]
Ah, of course! 
I think I got it wrong where you normally have to do:
```
connection.query((error, results) => {

});
```
But it doesn't happen much that I have some other params before the function.

Thanks for your answer. Kinda stupid mistake :(

**Comment by platinumazure** [comment]
@jessereitsma Glad to help! And don't worry, that sort of thing can definitely happen to anybody-- that's why we use linting tools! :smile:
